packages/app/src/components/inspector/ColorInputFieldWithPicker.tsx

```diff
@@ -145,49 +168,166 @@ export default memo(function ColorInputFieldWithPicker({
   id,
   value,
   onChange,
+  onChangeType,
+  onChangeGradientColor,
+  onChangeGradientPosition,
+  onAddGradientStop,
+  onChangeGradientType,
 }: Props) {
   // TODO: The value prop here can be an array, and other
   // inspector rows may also take arrays
-  const values = useMemo(() => [value], [value]);
-
-  const [state] = useApplicationState();
 
+  const [state, dispatch] = useApplicationState();
   const [swatchLayout, setSwatchLayout] = useState<SwatchLayout>('grid');
 
+  const values = useMemo(() => {
+    if (value._class === 'pattern')
+      //change to show preview of gradient :thinking_emoji:
+      return [];
+    return [value._class === 'gradient' ? value.stops[0].color : value];
+  }, [value]);
+
+  const selectedColor = values[0];
+
   const sharedSwatches = Selectors.getSharedSwatches(state);
 
+  const isSwatch = useMemo(
+    () =>
+      selectedColor &&
+      selectedColor.swatchID &&
+      sharedSwatches.some((e) => e.do_objectID === selectedColor.swatchID),
+    [selectedColor, sharedSwatches],
+  );
+
+  const detachThemeColor = useCallback(() => {
+    onChange({
+      ...selectedColor,
+      swatchID: undefined,
+    });
+  }, [onChange, selectedColor]);
+
+  const createThemeColor = useCallback(() => {
+    const swatchName = prompt('New Swatch Name');
+    if (!swatchName) return;
+
+    const id = uuid();
+    onChange({
+      ...selectedColor,
+      swatchID: id,
+    });
+    dispatch('addSwatch', swatchName, selectedColor, id);
+  }, [onChange, dispatch, selectedColor]);
+
+  const options: FillOption[] = useMemo(
+    () => [
+      'Solid Color',
+      'Linear Gradient',
+      'Radial Gradient',
+      'Angular Gradient',
+      'Pattern Fill',
+    ],
+    [],
+  );
+
+  const fillOption = useMemo(
+    () =>
+      value._class === 'color'
+        ? 'Solid Color'
+        : value._class !== 'gradient'
+        ? 'Pattern Fill'
+        : (`${
+            Sketch.GradientType[
+              value._class === 'gradient' ? value.gradientType : 0
+            ]
+          } Gradient` as FillOption),
+    [value],
+  );
+
   return (
     <Popover.Root>
       <Popover.Trigger as={Slot}>
-        <ColorInputField id={id} value={value} />
+        <ColorInputField
+          id={id}
+          value={value._class === 'color' ? values[0] : value}
+        />
       </Popover.Trigger>
       <Content side="bottom" align="center">
         <PaddedSection>
-          <ColorInspector
-            id={`${id}-panel`}
-            colors={values}
-            onChangeColor={onChange}
-          />
+          <Row>
```

Comment:
I know we need the `Popover.Root` and `Popover.Trigger` to stay mounted the whole time, but I wonder if we could split everything else in this `<Content>` element into 2 components? One for colors and one for gradients? I think it would help keep things organized. We might be able to move some of the logic above, e.g. `const fillOption = ...` into each component. We could leave the new components in this same file.

E.g. `ColorPickerContent`, `GradientPickerContent`

Reply:
👍

Reply:
I did it but the ContentComponents turns out too big and with a lot of props 🤔  I do think I can separate them in some way but I think at least keeping the fillOption outside would simplify this components. This fillOptions Logic it's used by all the FillType even patterns

Reply:
👍  I see what you mean. Maybe we leave it like this for now, and revisit when we add "Gradient Presets"

![Screenshot_6_14_21__8_57_AM](https://user-images.githubusercontent.com/1198882/121922277-a8ba5a80-ccee-11eb-88e7-7226a37ed7c0.png)

If we can separate out the swatch stuff now, I think that'll simplify this component a lot already